mocca-client/src/main/java/com/paypal/mocca/client/MoccaFeignEncoder.java

```diff
@@ -53,6 +70,17 @@ public void encode(Object object, Type bodyType, RequestTemplate template) throw
         }
     }
 
+    void validateVariables(List<Variable> vars) {
+        for (Variable var: vars) {
+            // should we skip "raw" variables?
+            Set<ConstraintViolation<Object>> violationSet = validator.validate(var.getValue());
```

Comment:
Here it seems that you are expecting the value to be a POJO whose instance attributes are annotated with BeanValidation annotations, correct?

If so, although that is a legitimate case, and should be supported, what about if the POJO itself is not annotated, but the method parameter is? Actually, but parameter could not be a POJO, and still be annotated with a Bean Validation annotation. I suppose this logic here won't be able to include this case, right?

Example:

``` java
public List<Book> getBooksByAuthor(@Var("authorName") @NotNull String authorName);
```

Reply:
Right. In this case because the string is not a bean it won't be validated. Do we want to handle this case? I don't know for sure if the `@NotNull` annotation is even passed to the encode method but I can check on that.

Reply:
Yes, we need to support this use case too. Not just this one, but all other validation annotations (besides `NotNull`) that could be applied to method parameters.